### DISCUSSION TITLE
Empty Result Set when calling stored procedure from code but not from SQL Developes

### DISCUSSION THREAD

**Discussion by AndreFK** [godror/godror#210] [Q&A] [answered]
I'm fairly new to golang, and I ran into a problem when trying to call a stored procedure.

``
QUALTRICS.QUALTRICS_DATA_PKG.fetch_email_survey_data
(
pi_date         IN varchar2 ,
pi_brand        IN varchar2,
pi_survey_type  IN varchar2,
po_survey_result OUT QUALTRICS_survey_data_typ
);
``

And this is the code I'm using from go

```
func (c *Client) LoadContactsFromDB(touchpoint string, date string, ctx context.Context){
     db, err := sql.Open("godror", connstr)
     stmt, err := db.Prepare("begin QUALTRICS.qualtrics_data_pkg.fetch_email_survey_data(:1,:2,:3,:4); end;")

     if err != nil {
             WriteLog("Statement Error: "+err.Error(), true)
	     return contacts
     }

     if err != nil {
	    WriteLog("Prepared Statement Error: "+err.Error(), true)
     }

     brand, survey := TouchpointToBrandAndSurveyType(touchpoint)

     WriteLog("Loading contacts from DB inserted on date 02/12/21 with touchpoint "+touchpoint+" ...", false)

     var out driver.Rows

     _, e := stmt.ExecContext(ctx, name, brand, survey, sql.Out{Dest: &out})

     rows, err := godror.WrapRows(ctx, db, out)

     for rows.Next(){
          fmt.Println("Found Row")
     }
}
```

The code doesnt't return any errors, and in fact I can see the columns's name from rows, but I dont get any data. It doesnt't get into that last for I have in my code. And when I go into SQLDeveloper and call the procedure from there with the same arguments, I get all the data I expect to get. Can someone tell me if I am doing something wrong, please?

**Comment by tgulacsi** [comment, marked as answer]
What is `QUALTRICS_survey_data_typ` ?
Your code does not check `e` of ExecContext and and `err` of WrapRows - is this intentional, error, or just the copy-paste?
What does `rows.Scan` return, and _most_importantly_ - what is `rows.Err()` at the end?
(BTW you want to `defer stmt.Close()` and `defer rows.Close()` to not leave cursors open).

**Reply by AndreFK** [reply]
I did't know that rows.Scan() would return an error message, but there I found the issue. One field was coming back as nil and I wasn't validating for it. It works perfectly know, thanks for helping me go in the right direction.
